yarl/_url.py

```diff
@@ -679,8 +682,7 @@ def raw_user(self) -> Union[str, None]:
 
         """
         # not .username
-        self._cache_netloc()
-        return self._cache["raw_user"]
+        return self._cached_split_netloc()[0]
```

Comment:
I don't really like the magic number here.. maybe a tuple unpack of named constant instead

Reply:
```diff
diff --git a/yarl/_url.py b/yarl/_url.py
index aea7132..abf6efc 100644
--- a/yarl/_url.py
+++ b/yarl/_url.py
@@ -682,7 +682,8 @@ class URL:
 
         """
         # not .username
-        return self._cached_split_netloc()[0]
+        raw_user, _, _, _ = self._cached_split_netloc()
+        return raw_user
 
     @cached_property
     def user(self) -> Union[str, None]:
@@ -702,7 +703,8 @@ class URL:
         None if password is missing.
 
         """
-        return self._cached_split_netloc()[1]
+        _, raw_password, _, _ = self._cached_split_netloc()
+        return raw_password
 
     @cached_property
     def password(self) -> Union[str, None]:
@@ -726,7 +728,8 @@ class URL:
         """
         # Use host instead of hostname for sake of shortness
         # May add .hostname prop later
-        return self._cached_split_netloc()[2]
+        _, _, raw_host, _ = self._cached_split_netloc()
+        return raw_host
 
     @cached_property
     def host(self) -> Union[str, None]:
@@ -781,7 +784,8 @@ class URL:
         None for relative URLs or URLs without explicit port.
 
         """
-        return self._cached_split_netloc()[3]
+        _, _, _, explicit_port = self._cached_split_netloc()
+        return explicit_port
 
     @cached_property
     def raw_path(self) -> str:
```

Thats really not much better though

Reply:
https://github.com/aio-libs/yarl/pull/1339 ... maybe.. but than we still need to lookup the value